sharedbuffers/mapped_struct.py

```diff
@@ -776,7 +788,12 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0,
                         # longs are tricky, give up
                         all_int = all_intlong = 0
         if all_int or all_intlong:
-            if dtype == b'l' or dtype == b'L':
+            l_or_L = False
+            if check_udtype:
+                l_or_L = udtype[0] == u'l' or udtype[0] == u'L'
+            else:
+                l_or_L = dtype == b'l' or dtype == b'L'
```

Comment:
There are cases in py3 where you can end up here. Wouldn't that fail the comparison because of unicode-bytes mismatch?

Maybe always set udtype and instead of "check_udtype" just use "if python3"